src/session/client-certificate.c

```diff
@@ -297,16 +297,12 @@ cockpit_session_client_certificate_map_user (const char *client_certificate_file
   if (read_cert_file (client_certificate_filename, cert_pem, sizeof cert_pem) < 0)
     {
       warnx ("No https instance certificate present");
-      return NULL;
+      exit_init_problem ("authentication-unavailable", "No https instance certificate present");
     }
 
   size_t ws_cgroup_length;
   char *ws_cgroup = read_proc_self_cgroup (&ws_cgroup_length);
-  if (ws_cgroup == NULL)
-    {
-      warnx ("Could not determine cgroup of this process");
-      return NULL;
-    }
+  assert (ws_cgroup);
   /* A simple prefix comparison is appropriate here because ws_cgroup
    * will contain exactly one newline (at the end), and the expected
```

Comment:
I almost feel like *this* is the place to make the extra check to ensure that ws_cgroup definitely ends with `\n`.

Reply:
#16808 changes this a lot, so I'd prefer to make the validation changes there.